include/model.php

```diff
@@ -695,6 +695,54 @@ public function filter_language_terms_orderby( $orderby, $args, $taxonomies ) {
 		return sprintf( '%1$s.term_group, %1$s.term_id', $matches['alias'] );
 	}
 
+	/**
+	 * Adds the missing language terms for the secondary language taxonomies.
+	 *
+	 * @since 3.4
+	 *
+	 * @return void
+	 */
+	public function add_missing_secondary_language_terms() {
+		foreach ( $this->get_languages_list() as $language ) {
+			$this->update_secondary_language_terms( $language->slug, $language->name, $language );
+		}
+
+		// Clear the cache, so the new `term_id` and `term_taxonomy_id` appear in the languages list.
+		$this->clean_languages_cache();
+	}
+
+	/**
+	 * Updates or adds new terms for a secondary language taxonomy (aka not 'language').
+	 *
+	 * @since 3.4
+	 *
+	 * @param string            $slug     Language term slug (with or without the `pll_` prefix).
+	 * @param string            $name     Language name (label).
+	 * @param PLL_Language|null $language Optional. A language object. Required to update the existing terms.
+	 * @return void
+	 *
+	 * @phpstan-param non-empty-string $slug
+	 * @phpstan-param non-empty-string $name
+	 */
+	protected function update_secondary_language_terms( $slug, $name, PLL_Language $language = null ) {
+		$slug = 0 === strpos( $slug, 'pll_' ) ? $slug : "pll_$slug";
+
+		foreach ( $this->translatable_objects->get_secondary_translatable_objects() as $object ) {
+			if ( ! empty( $language ) ) {
+				$term_id = $language->get_tax_prop( $object->get_tax_language(), 'term_id' );
+			} else {
+				$term_id = 0;
+			}
+
+			if ( ! empty( $term_id ) ) {
+				wp_update_term( $term_id, $object->get_tax_language(), array( 'slug' => $slug, 'name' => $name ) );
```

Comment:
I thought that WP was comparing old and new terms before the update, but it seems that's not the case so maybe we should do it ourselves to avoid useless code execution (this part of the code may be called more frequently now than before). 
```suggestion
				if ( $language->slug !== $slug || $language->name !== $name ) {
					wp_update_term( $term_id, $object->get_tax_language(), array( 'slug' => $slug, 'name' => $name ) );
				}
```

Reply:
Done in [e21f962](https://github.com/polylang/polylang/pull/1176/commits/e21f962a5b5fd4ef37e2208242f65f8356da728c).
I tweaked a bit your suggestion to prevent adding another floor to the pyramid.